lib/http-bridge/test/integration/Cardano/Wallet/HttpBridge/NetworkSpec.hs

```diff
@@ -144,6 +155,27 @@ spec = do
                     "Transaction failed verification: transaction has more \
                     \witnesses than inputs"
             runExceptT (postTx bridge signed) `shouldReturn` err
+
+    describe "waitForConnection" $ do
+        it "times out after a short while" $ do
+            nw <- newNetworkLayer
+            waitForConnection nw 2 `shouldThrow` \case
+                ErrNetworkTipNetworkUnreachable _ -> True
+                _ -> False
+
+        it "returns when the network becomes availible" $ do
+            c <- newNetworkLayer
+            handle <- async . launch . return
+                $ httpBridgeWithSetup (threadDelay 3000000)
+            -- Start the bridge after 3s, and make sure waitForConnection
+            -- returns within (3s + 3s extra)
```

Comment:
Why do we need this extra 3s ? Can't we just start the bridge and call wait for connection.. That's how we would normally use it, isn't it ?

Reply:
This was to check that `waitForConnection` did in-fact return within a reasonable time, not just that it returned. Overkill?

Reply:
Hmmmm. I still don't get it? Also, are we sure that 3 seconds is indeed enough in every environment? I'd rather just wait for connection, and verify that it resolve. If it always return within 3 seconds, then why wait for `20s` in the implementation? If we are ready to wait up-to 20s that's what we should be testing here and this should never throw.